official/resnet/ctl/ctl_imagenet_main.py

```diff
@@ -0,0 +1,297 @@
+          loss2 = (tf.reduce_sum(model.losses) /
+                   tf.distribute.get_strategy().num_replicas_in_sync)
+          loss = loss1 + loss2
+
+        grads = tape.gradient(loss, model.trainable_variables)
+        optimizer.apply_gradients(zip(grads, model.trainable_variables))
+
+        training_accuracy.update_state(labels, logits)
+        return loss
+
+      if strategy:
+        per_replica_losses = strategy.experimental_run_v2(
+            step_fn, args=(train_ds_inputs,))
+        return strategy.reduce(tf.distribute.ReduceOp.SUM, per_replica_losses,
+                               axis=None)
+      else:
+        return step_fn(train_ds_inputs)
+
+    def test_step(test_ds_inputs):
+      """Evaluation StepFn."""
+      def step_fn(inputs):
+        images, labels = inputs
+        logits = model(images, training=False)
+        loss = tf.keras.losses.sparse_categorical_crossentropy(labels,
+                                                               logits)
+        loss = tf.reduce_sum(loss) * (1.0/ flags_obj.batch_size)
+        test_loss.update_state(loss)
+        test_accuracy.update_state(labels, logits)
+
+      if strategy:
+        strategy.experimental_run_v2(step_fn, args=(test_ds_inputs,))
+      else:
+        step_fn(test_ds_inputs)
+
+    if flags_obj.use_tf_function:
+      train_step = tf.function(train_step)
+      test_step = tf.function(test_step)
+
+    time_callback.on_train_begin()
+    for epoch in range(train_epochs):
+
+      train_iter = iter(train_ds)
+      total_loss = 0.0
+      training_accuracy.reset_states()
+
+      for step in range(train_steps):
+        optimizer.lr = keras_imagenet_main.learning_rate_schedule(
+            epoch, step, train_steps, flags_obj.batch_size)
+
+        time_callback.on_batch_begin(step+epoch*train_steps)
+        total_loss += train_step(next(train_iter))
+        time_callback.on_batch_end(step+epoch*train_steps)
+
+      train_loss = total_loss / train_steps
+      logging.info('Training loss: %s, accuracy: %s%% at epoch: %d',
+                   train_loss.numpy(),
+                   training_accuracy.result().numpy(),
+                   epoch)
+
+      if (not flags_obj.skip_eval and
+          (epoch + 1) % flags_obj.epochs_between_evals == 0):
+        test_loss.reset_states()
+        test_accuracy.reset_states()
+
+        test_iter = iter(test_ds)
+        for _ in range(eval_steps):
+          test_step(next(test_iter))
+
+        logging.info('Test loss: %s, accuracy: %s%% at epoch: %d',
+                     test_loss.result().numpy(),
+                     test_accuracy.result().numpy(),
+                     epoch)
+
+    time_callback.on_train_end()
+
+    eval_result = None
+    train_result = None
+    if not flags_obj.skip_eval:
+      eval_result = [test_loss.result().numpy(),
+                     test_accuracy.result().numpy()]
+      train_result = [train_loss.numpy(),
```

Comment:
nit: setting train_result can be outside the if block?

Reply:
True but that would mean useless values would be populated when we don't run the Accuracy test. Hence moved it together with the eval results.